### ISSUE TITLE
weight NCHW2NHWC

### ISSUE THREAD

**Issue by zhly0** [tensorflow/tensorflow#30044]
Hi，I know that if I want to change the tensor from NCHW to NHWC is simply use transpos((0,2,3,1)) will,but my question is what if I want to change the weight of network trained using NCHW to NHWC,is using  transpos((0,2,3,1)) on the the weight will work too?I just want to know.Thanks if anyone can help！

**Comment by zhly0** [comment]
It is weird that I use this code:
`#-*- coding:utf-8 -*-
import tensorflow as tf

input_x = tf.constant([
	[
		[[0.0, 1.0, 2.0],[1.0,1.0,0.0],[1.0,1.0,2.0],[2.0,2.0,0.0],[2.0,0.0,2.0]],
		[[0.0,0.0,0.0],[1.0,2.0,0.0],[1.0,1.0,1.0],[0.0,1.0,2.0],[0.0,2.0,1.0]],
		[[1.0,1.0,1.0],[1.0,2.0,0.0],[0.0,0.0,2.0],[1.0,0.0,2.0],[0.0,2.0,1.0]],
		[[1.0,0.0,2.0],[0.0,2.0,0.0],[1.0,1.0,2.0],[1.0,2.0,0.0],[1.0,1.0,0.0]],
		[[0.0,2.0,0.0],[2.0,0.0,0.0],[0.0,1.0,1.0],[1.0,2.0,1.0],[0.0,0.0,2.0]],
	],
])
filters = tf.constant([
	[
		[[1.0,-1.0,0.0],[1.0,0.0,1.0],[-1.0,-1.0,0.0]],
		[[-1.0,0.0,1.0],[0.0,0.0,0.0],[1.0,-1.0,1.0]],
		[[-1.0,1.0,0.0],[-1.0,-1.0,-1.0],[0.0,0.0,1.0]],
	],
])
bias = tf.constant(1.0,shape=[1])

filterinput = tf.reshape(filters, [3, 3, 3, 1])  # 将原始的filter转换成适合tf的输入的形式
print('input_x shape',input_x.shape)
print('filterinput shape',filterinput.shape)
res = tf.nn.conv2d(input_x, filterinput, strides=[1,2,2,1], padding='SAME')#+bias

sess = tf.Session()
print("result shape:", res.shape)
print(sess.run(res[:,:,:,0]))
`
get a result:
`input_x shape (1, 5, 5, 3)
filterinput shape (3, 3, 3, 1)
result shape: (1, 3, 3, 1)
[[[ 0. -1. -4.]
  [-7.  0.  0.]
  [ 3. -4.  0.]]]
`
and I transpose the input to NCHW,and weight from RSCK to KCRS(R:kernel h,S:kernel w,C:in channels,K:out channels):
`#-*- coding:utf-8 -*-
import tensorflow as tf
import os
#os.environ['CUDA_VISIBLE_DEVICES']='-1'
input_x = tf.constant([
	[
		[[0.0, 1.0, 2.0],[1.0,1.0,0.0],[1.0,1.0,2.0],[2.0,2.0,0.0],[2.0,0.0,2.0]],
		[[0.0,0.0,0.0],[1.0,2.0,0.0],[1.0,1.0,1.0],[0.0,1.0,2.0],[0.0,2.0,1.0]],
		[[1.0,1.0,1.0],[1.0,2.0,0.0],[0.0,0.0,2.0],[1.0,0.0,2.0],[0.0,2.0,1.0]],
		[[1.0,0.0,2.0],[0.0,2.0,0.0],[1.0,1.0,2.0],[1.0,2.0,0.0],[1.0,1.0,0.0]],
		[[0.0,2.0,0.0],[2.0,0.0,0.0],[0.0,1.0,1.0],[1.0,2.0,1.0],[0.0,0.0,2.0]],
	],
])
filters = tf.constant([
	[
		[[1.0,-1.0,0.0],[1.0,0.0,1.0],[-1.0,-1.0,0.0]],
		[[-1.0,0.0,1.0],[0.0,0.0,0.0],[1.0,-1.0,1.0]],
		[[-1.0,1.0,0.0],[-1.0,-1.0,-1.0],[0.0,0.0,1.0]],
	],
])
bias = tf.constant(1.0,shape=[1])

filterinput = tf.reshape(filters, [3, 3, 3, 1])
print('input_x shape',input_x.shape)
print('filterinput shape',filterinput.shape)
input_x = tf.transpose(input_x,perm=(0,3,1,2))
filterinput = tf.transpose(filterinput,perm=(2,3,1,0))
print('input_x shape',input_x.shape)
print('filterinput shape',filterinput.shape)
res = tf.nn.conv2d(input_x, filterinput,data_format='NCHW',strides=[1,1,2,2], padding='SAME')#+bias
res = tf.transpose(res,perm=(0,3,2,1))
#res = tf.reduce_sum(res, 3, keep_dims=True)
sess = tf.Session()
print("result shape:", res.shape)
print(sess.run(res[:,:,:,:]))`
and I get this:
`input_x shape (1, 5, 5, 3)
filterinput shape (3, 3, 3, 1)
input_x shape (1, 3, 5, 5)
filterinput shape (3, 1, 3, 3)
result shape: (1, 3, 3, 3)
[[[[-2. -2. -1.]
   [-2.  2.  2.]
   [-1.  1. -3.]]

  [[-2.  0.  0.]
   [ 0.  1. -1.]
   [-1.  0. -3.]]

  [[-2. -1.  1.]
   [ 1.  1. -5.]
   [ 0. -3. -2.]]]]`
I same inputs and kernels,the dfierent is NCHW and NHWC,the output shape and result is different.why?????
